sklearn/metrics/cluster/tests/test_supervised.py

```diff
@@ -172,6 +172,20 @@ def test_expected_mutual_info_overflow():
     assert expected_mutual_information(np.array([[70000]]), 70000) <= 1
 
 
+def test_int_overflow_mutual_info_score():
+    # Test overflow in mutual_info_classif
+    x = np.concatenate((np.repeat(1, 52632 + 2529), np.repeat(2, 14660+793),
+                       np.repeat(3, 3271+204), np.repeat(4, 814+39),
+                       np.repeat(5, 316+20)))
+    y = np.concatenate((np.repeat(0, 52632), np.repeat(1, 2529),
+                       np.repeat(0, 14660), np.repeat(1, 793),
+                       np.repeat(0, 3271), np.repeat(1, 204),
+                       np.repeat(0, 814), np.repeat(1, 39),
+                       np.repeat(0, 316), np.repeat(1, 20)))
+
+    mutual_info_score(x.ravel(), y.ravel())
```

Comment:
Did you forgot to assert the score to be sure that we don't have `NaN`?

Reply:
No, this check is only to be assured there's no overflow. Why do I check for a NaN?

Reply:
Previously, it was already possible to call the function and it was resulting to nan (due to the overflow in the log). You should at least check that the output is finite to be sure that they is no overflow. Otherwise, your current test is also passing in the current master branch,

Reply:
The test does not pass in 32 bit Python in master branch but passes otherwise. I'll soon add the suggested check.